### PR TITLE
python310Packages.apptools: Enable more tests

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Would probably be worth opening this as a PR upstream and then doing a `fetchpatch` of that.

**Comment by onny** [comment]
Thank you @risicle, fixed it :)

File: pkgs/development/python-modules/apptools/default.nix [hunk 1]
```diff
@@ -26,11 +26,20 @@ buildPythonPackage rec {
   };
 
   patches = [
+    # python310: Fix tests
+    # https://github.com/enthought/apptools/issues/303
+    (fetchpatch {
+      url = "https://github.com/enthought/apptools/commit/0d65869701b3bab49d253c5489002664316b32fb.patch";
```
Comment (risicle) [inline pkgs/development/python-modules/apptools/default.nix:32]: https://github.com/enthought/apptools/commit/10fb73916124f7ae7edf6c6688a05ad95678488f is the commit that actually made it into upstream - think they probably did a rebase-merge of the PR.

Comment (onny) [reply pkgs/development/python-modules/apptools/default.nix]: I updated the fetchpatch commit to the one being merged upstream :)

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**